### ISSUE TITLE
Issue generating coverage with AttributeFilter

### ISSUE THREAD

**Issue by manemarron** [SteveGilham/altcover#28]
I have many test projects in my solution so I am using AltCover as my coverage calculator for each project and ReportGenerator (https://github.com/danielpalme/ReportGenerator) as my coverage aggregator. When I run 

```bash
$  dotnet test /p:AltCover=true /p:AltCoverAttributeFilter=ExcludeFromCodeCoverage
$ dotnet reportgenerator "-reports:coverage.xml" "-targetdir:coverage" "-reporttypes:Html;Xml"
```

on any project, the coverage file is generated but ReportGenerator can't parse it (it throws an Object Reference exception). When I remove the attribute filter, everything works fine.

I am using the latest version of altcover and reportgenerator-cli
I have reported the issue both here and in https://github.com/danielpalme/ReportGenerator/issues/146

**Comment by SteveGilham** [comment]
I have a what looks like repro for this (my current development system, and ReportGenerator 3.1.2)
```
Unhandled Exception: System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Palmmedia.ReportGenerator.Parser.OpenCoverParser.<>c.<ProcessClass>b__13_5(XElement m)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Palmmedia.ReportGenerator.Parser.OpenCoverParser.ProcessClass(Dictionary`2 fileIdsByFilename, Assembly assembly, String className)
...
```
In the generated coverage report, a filtered method looks like
```xml
            <Method skippedDueTo="Filter" visited="false" cyclomaticComplexity="1" nPathComplexity="0" sequenceCoverage="0" branchCoverage="0" isConstructor="false" isStatic="true" isGetter="false" isSetter="false" crapScore="0">
              <Summary numSequencePoints="0" visitedSequencePoints="0" numBranchPoints="0" visitedBranchPoints="0" sequenceCoverage="0" branchCoverage="0" maxCyclomaticComplexity="0" minCyclomaticComplexity="0" visitedClasses="0" numClasses="0" visitedMethods="0" numMethods="0" minCrapScore="0" maxCrapScore="0" />
              <MetadataToken>100663299</MetadataToken>
              <Name>System.Void N.DU::testMakeUnion()</Name>
              <FileRef />
              <SequencePoints />
              <BranchPoints />
              <MethodPoint />
            </Method>
```

Editing the report file to remove the empty `FileRef` tag and re-running the report generation worked OK.

tl;dr it's a Postel's Principle fail.  I'm not emitting conservatively enough.

**Comment by danielpalme** [comment]
I'm currently expecting that the `<FileRef />` element contains an `uid` attribute. Can you supply this even if the method is filtered?

**Comment by SteveGilham** [comment]
It's actually simpler for me to omit the tag entirely for methods that aren't being instrumented, and the ReportGenerator seems happy enough with that, too.

**Comment by SteveGilham** [comment]
[Pre-release build available for testing](https://ci.appveyor.com/project/SteveGilham/altcover/build/3.5.576-pre/artifacts) -- otherwise it'll be in the next release

**Comment by SteveGilham** [comment]
And formally in release 3.5.580
